pkg/scheduler/internal/cache/cache.go

```diff
@@ -256,7 +256,7 @@ func (cache *schedulerCache) UpdateSnapshot(nodeSnapshot *Snapshot) error {
 		nodeSnapshot.generation = cache.headNode.info.Generation
 	}
 
-	if len(nodeSnapshot.nodeInfoMap) > len(cache.nodes) {
+	if len(nodeSnapshot.nodeInfoMap) > cache.nodeTree.numNodes {
```

Comment:
can you add a comment explaining that deleted nodes get removed from the tree but are kept in the cache until the pods are deleted, and hence here we check against the number of nodes in the tree not the cache.

Reply:
also, document that the snapshot (both map and lists) only include nodes that are not deleted as of the time the snapshot was taken, and so nodeinfo.Node() is guaranteed not to be nil.

Reply:
Done. I amended.